lib/pages/settings/view/request_card.dart

```diff
@@ -0,0 +1,213 @@
+                            .headline6
+                            .copyWith(fontSize: 14),
+                      ),
+                      const SizedBox(height: 10),
+                      _buildButtons(request.processed, adminProvider, request)
+                    ],
+                  ),
+                ),
+              ),
+            );
+          } else {
+            return const SizedBox.shrink();
+          }
+        });
+  }
+
+  Widget _buildUserHeader(String userId, bool processed, bool accepted,
+          AdminProvider adminProvider) =>
+      FutureBuilder(
+          future: adminProvider.fetchUserById(userId),
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.done) {
+              final User user = snapshot.data;
+              return Column(
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: [
+                        Flexible(
+                          child: Text(
+                            '${user?.firstName ?? 'Unknown User'} ${user?.lastName ?? ''}',
+                            style: Theme.of(context)
+                                .textTheme
+                                .headline6
+                                .copyWith(
+                                    fontSize: 18, fontWeight: FontWeight.bold),
+                            overflow: TextOverflow.ellipsis,
+                            maxLines: 2,
+                          ),
+                        ),
+                        if (processed == true && accepted != null)
+                          _buildAcceptedMarker(accepted),
+                      ],
+                    ),
+                    Text(
+                      '${user?.classes != null ? user?.classes[user.classes.length - 1] : '-'}',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline6
+                          .copyWith(fontSize: 14, fontWeight: FontWeight.bold),
+                      overflow: TextOverflow.fade,
+                      maxLines: 2,
+                    ),
+                  ]);
+            } else {
+              return Column(
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Text(
+                      '-',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline6
+                          .copyWith(fontSize: 18),
+                    ),
+                    Text(
+                      '',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline6
+                          .copyWith(fontSize: 14),
+                    ),
+                  ]);
```

Comment:
I don't think you need to duplicate the code here.

You could try something like:

```
User user;
if (snapshot.connectionState == ConnectionState.done) {
              user = snapshot.data;
}
```
at the beginning of the function, and then in the text have something like
```
connectionState != ConnectionState.done ? '-' : '${user?.firstName ?? 'Unknown User'} ${user?.lastName ?? ''}'
```